internal/git/gitsync_processor.go

```diff
@@ -226,6 +274,64 @@ func ApplyPatchToResources(patchedResources PatchedResource, kubeClient kubernet
 	return nil
 }
 
+// CheckRepoUpdatesForKustomize will calculate for any update required based on commit history.
+func CheckRepoUpdatesForKustomize(ctx context.Context, r *git.Repository, repo *v1alpha1.RepositoryPath, lastCommitHash, defaultNameSpace string, k kustomize.Kustomize) (PatchedResource, string, error) {
+	var patchedResources PatchedResource
+	var recentHash string
+
+	logger := logging.FromContext(ctx).With("RepositoryPath Name", repo.Name, "Repository Url", repo.RepoUrl)
+	// Build the kustomize manifest with old changes.
+	oldManifest, err := k.Build(nil)
+	if err != nil {
+		logger.Errorw("error generating kustomize manifest", "err", err)
+		return patchedResources, recentHash, err
+	}
+
+	if err = fetchUpdates(r); err != nil {
+		logger.Errorw("error checking for updates in the github repo", "err", err)
+		return patchedResources, recentHash, err
+	}
+
+	remoteRef, err := getLatestCommitHash(r, repo.TargetRevision)
+	if err != nil {
+		logger.Errorw("failed to get latest commits in the github repo", "err", err)
+		return patchedResources, recentHash, err
+	}
+
+	// If remote commit hash and local one aren't equal, it means there are new changes made in remote repository.
+	if remoteRef.String() != lastCommitHash {
```

Comment:
just to verify, if the kustomization.yaml references an upstream yaml from another repo, we won't detect that, right? I think what @xdevxy had in mind was that we recalculate the manifests every time, to prevent not seeing when that changed. But I'm okay worrying about this in a subsequent PR.

Reply:
Right, currently we don't detect kustomization reference to another repo, I can create an issue if not already open.

Reply:
yes please, and probably assign it to yourself? 
And maybe we should compare the two options of either: a. watching the upstream repositories for changes; b. not watching but just performing the kustomization change every cycle

Reply:
Created it #100 i'll add more detail to it